fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/version/GetLatestVersionDAO.java

```diff
@@ -65,7 +65,16 @@ public GetLatestVersionDAO(String adminSchemaName, String schemaName) {
                 int version = rs.getInt(4);
 
                 String schemaTypeName = schema + ":" + type + ":" + name;
-                result.put(schemaTypeName, version);
+                result.compute(schemaTypeName, (key, currentValue) -> {
```

Comment:
basically you are reconciling conflicts in the Version_HISTORY? 

Wouldn't it be better to reconcile in the sql? What if this table is migrated and the restored insert order is different?

Reply:
what its doing:  if we already have an entry for this `shema:type:name` in our map (`currentValue != null`), then that means that we have two entries that only vary by case (which is the root of the issue).

so, in that case, we log a warning and then take the higher of the two versions.

for example, consider there are two entries in the table:
|schema|type|name|version|
|--|--|--|--|
my_schema | TABLE | MY_TABLE | 6
MY_SCHEMA | TABLE | MY_TABLE | 8

because we UCASE them above, then both become `MY_SCHEMA:TABLE:MY_TABLE`.
previously we were always overwriting the map entry which is non-deterministic.  when it came back 6, we were trying to reapply the 6->8 migration which failed.
with this change, we will always use the higher value, which will prevent us from trying the same migration over and over, even when the table contains 2 entries for a single schema name (due to the casing issue described in #1971)

Reply:
This should be done in the DAO query. The query should select upper(schema_name). This also needs to be added to the group by: 
```        final String cols = DataDefinitionUtil.join("UPPER(" + SchemaConstants.SCHEMA_NAME + ")", SchemaConstants.OBJECT_TYPE, SchemaConstants.OBJECT_NAME);
```

Reply:
I would also UPPER(object_name) too, just to be on the safe-side, because: 
```
                String name = rs.getString(3).toUpperCase();
```

Reply:
Although having the warning message is useful, which you wouldn't get by changing the query. So I'd just suggest mentioning something about "case" in the warning message, to remind us what's going on the next time this crops up.

Reply:
OK, I added "Check schema name casing." to the end of the warning message.  If that sounds good, please approve the PR and we can close out this issue.

Reply:
> This should be done in the DAO query. The query should select upper(schema_name). This also needs to be added to the group by:
> 
> ```
>  final String cols = DataDefinitionUtil.join("UPPER(" + SchemaConstants.SCHEMA_NAME + ")", SchemaConstants.OBJECT_TYPE, SchemaConstants.OBJECT_NAME);
> ```

@punktilious I thought we did this and decided it wasn't the right approach.  I had used UCASE in the query maybe a few months back.  I even added the toUpperCase.  

I do think we might be rehashing.

Reply:
> I do think we might be rehashing.
Partially my fault; just trying to be clear on the desired behavior.

The advantage to doing it in the java code here is we can give a nice warning message.  However, what isn't clear to me is what the expected user action should be when they see this message.

If we want them to just ignore it, then I’m thinking we should either 
A. just fix the query / revert my proposed change; or
B. change the log message to info level (or lower)

If we want them to fix their version history table up, then I should probably update the warning message to indicate that (or point to somewhere that we document that this is safe for them to do).  The message would become something like
> To avoid this warning, please normalize the schema and object names to UPPER case in FHIR_ADMIN.VERSION_HISTORY.

Which one do we want?

Reply:
team decision:  we don't want to ask the user to clean up the VERSION_HISTORY table at this time, and so I will reduce the log level of this message of FINE